### DISCUSSION TITLE
AT32F435 fw 3.36 hangs on specific directory

### DISCUSSION THREAD

**Discussion by lyzanxia** [keirf/flashfloppy#700] [Q&A]
I think I am obviously doing something wrong... 
But when I flash 3.36 on my AT32F435 gotek, it works fine until next reboot (and the image_a file is getting created).
The oled garbles and I see the drive light flicker, but nothing gets read. If I delete image_a, everything works again until next reboot.
Flashing back to 3.35 fixes everything.

Strapped the MOR jumper, but do not know which one the MCU jumper is?
(Put interface amiga and drive delay 500)..

**Comment by keirf** [comment]
IMAGE_A gets created whenever a new disk image is selected on the Gotek.

You might have to give more details. I did a quick full flash to 3.35, and then updated to 3.36, and selected a few image files on my USB stick. No problems.

So what were your steps? What is your FF.CFG? Etc etc

**Reply by keirf** [reply]
Also I don't know what you are referring to when you mention the "MCU jumper"?

**Reply by lyzanxia** [reply]
I did some more troubleshooting. If I change interface from shugart to amiga in ff.cfg I experience these problems.
Tried different USB sticks. Have attached the ff.cfg. If needed I can make a video of the problem.

Steps where upgrade from 3.34 to 3.36, changed interface shugart to amiga (had motor delay 500 already, since I had MOR already strapped). On first power on/off of the Amiga after flash, display show flashfloppy+ 3.36 and thereafter garbage on screen and gotek green light flickers a few times.

Sorry I misread "MCU jumper" as in : "Requires AT32F435 "MCU and MOR jumper strap".. :)

[FF.txt](https://github.com/keirf/flashfloppy/files/9808503/FF.txt)

**Reply by keirf** [reply]
Interesting. This is just while running the Gotek as DF0 in Amiga Workbench? What Kickstart version?

**Reply by lyzanxia** [reply]
Yes, it is the only drive in the amiga (df0).
Just powering up, gives a garbled oled display and gotek drive light flickers a few times and then kicks up the kickstart screen. 
Here is a (bad) video of the display after powering up, first time it shows the directory (91 is the year of megademo dir), I then reset again and it show garbage..

kickstart is 1.3, rev6a motherboard.

https://user-images.githubusercontent.com/59722594/196389914-73455874-84b2-4008-8742-4f033f255a61.mp4

**Reply by keirf** [reply]
Oooook. So, no high-density (1760kB) ADF images here? We're just talking 880kB images, and as soon as you select an image at Kickstart screen, badness happens? Like, you can't even boot an image using 3.36, at all?

**Reply by keirf** [reply]
And jumpers at S0 and MOR only? Nothing at JC for example?

**Comment by keirf** [comment]
I am running 3.36 on an AT32F435 Gotek with jumpers at MOR and S0. I am running with your FF.CFG. And I have a couple of bootable 880kB (double density) ADF images: one in the root, and one in a folder called 91.

I am running as DF0 on an A500 running KS1.3.

I can reset the Amiga and select images, over and over, and it works okay for me.

Can you think of any differences in your setup? For example, do you have a particular USB stick setup? Deep folder hierarchy?

I will also add that resetting the Amiga should not directly affect the Gotek. The Gotek doesn't get reset by Amiga reset. In fact the Amiga reset line is not physically connected to the Gotek.

**Reply by keirf** [reply]
The presence of IMAGE_A vs not makes me think this has something to do with the name or location of the image you are selecting. Or perhaps the image itself. Is that varying at all, or are you always duplicating the contents of the USB stick onto a new one and selecting the same image file in the same location with the same name?

**Reply by lyzanxia** [reply]
So during my lunch break, tried some more. S0+MOR are only jumpered.

You are right, it's not during a reset, it's power off/on.

It seems you are on the right track in thinking about location. If I choose a directory of WB1.3 and mount an image inside it's ok.
With ok, I mean the IMAGE_A gets created and I turn off/on the amiga and it still shows the last dir on oled and loads the image selected.

The "problem" dir is : Photons-ADF-Demo-Disk-Archive\ALL-megademos-87-95\91

Did not had the time to try if it is the depth ,capitals ,the minus sign in the name etc..
Maybe directory properties or anything...

The usb sticks I try are indeed all copies (file copy - fat32).

Will try some more later on..thank you for suggestions so far!

**Reply by keirf** [reply]
Excellent feedback. The aim is that I can reproduce this. It's probably a memory corruption bug, that's probably been there a while and revealed only in the 3.36 build for some reason. Basically really hard to fix unless I can reproduce it, at which point I can dig into it properly.

**Reply by keirf** [reply]
It could also be the number of files, or aggregate total length of filenames, inside a folder. I bet this demo distribution is pretty big for 1991! Is it downloadable somewhere? I could stick it on a USB stick myself, exactly as it comes, which I assume is what you've done.

**Reply by keirf** [reply]
The thing is, the old Goteks had limited memory, and FlashFloppy would give up sorting a folder if it doesn't fit in RAM. 435 has *loads* more RAM. There's plenty scope for a memory management or sorting or indexing bug of some kind to bite when a folder is biiig.

**Reply by lyzanxia** [reply]
Sure, it's here : http://coppershade.org/articles/More!/Downloads/  , it's great by the way :).
Hope you can get it reproduced. 

Later on I will try to some more (in changing dir names/changing depth etc..)

**Reply by keirf** [reply]
Well that sounded promising. I unzipped the archive to my stick and went and navigated to subfolder 91 no problem. I can select any image in there, eject, insert, etc etc. No problem. I can turn power off, turn it on again, and state seems to be restored okay...

Is there a particular image selected or navigation point you are at when things go wrong? Is it simply the act of selecting subfolder 91 that guarantees your Gotek's demise?

**Reply by lyzanxia** [reply]
I just tried to make a new usb stick..only photons archive and a ff.cfg,image_a and the .upd are extra on this stick and error is here.
Maybe if you try to put these files beside the archive you can reproduce it?
[FF.zip](https://github.com/keirf/flashfloppy/files/9810073/FF.zip)

**Reply by keirf** [reply]
Okay, let's see if your Gotek's memory is reliable. Try the attached firmware update. It is a memory tester. As long as the number on first row keeps counting up, all is good. If it stops, you see an error report.

[mem-test.zip](https://github.com/keirf/flashfloppy/files/9810191/mem-test.zip)

(You can revert to normal firmware via update method, same as usual)

PS. I have now tried with FF.CFG, IMAGE_A and the UPD file also on the stick. I still cannot reproduce the error.

**Reply by lyzanxia** [reply]
Memory test keeps counting for about 20 mins now, so guess this is OK.

**Reply by keirf** [reply]
What a damn mystery then. It will be interesting to see if you can drill down later on exactly what does vs doesn't cause the problem for you, I suppose. Could also try a memory test with delay between fill and check (to allow memory bits to flip). It's not like it's DRAM though. And the whole "it's a memory problem" theory is a wild stab anyway.

**Reply by keirf** [reply]
Is yours the model with factory OLED, or is your OLED retrofitted?

**Reply by lyzanxia** [reply]
Pfew, I got it...as simple as read only permissions on this photon archive folder. (no idea how they got there).
If I clear readonly permission on this dir in windows, all ok..if I enable it, 3.36 hangs up (maybe 3.35 ignores this or something?).

**Reply by lyzanxia** [reply]
It is a retrofitted oled (one of the 3 drives I got from you :) )

**Reply by keirf** [reply]
So, literally, you removed read-only attribute from the 91 folder? And that's it??

**Reply by lyzanxia** [reply]
Removed it from the parent dir. To do a sanity test , I afterwards enabled it again and error was back.
I guess if you enable read only on the parent Photon dir, you will hopefully get the error also..
![Capture](https://user-images.githubusercontent.com/59722594/196443073-c002507a-f0df-4fd6-ac85-99936289da3a.JPG)

**Reply by lyzanxia** [reply]
Just fixed my other amiga's usb stick with the same solution. They all work fine now on 3.36 with interface=amiga and motor-delay 500.

**Reply by keirf** [reply]
I wonder if it's dependent on your Windows version and/or how you unzipped the archive. On Windows 11 the R/O attrib doesn't seem to be plumbed through to File Explorer. When I use `attrib` at DOS prompt I do not see the R/O attribute on any folder. If I add it to the 91 subfolder I still do not reproduce the error. Perhaps there were other (bad) attributes applied to that folder, or something? Perhaps see what `attrib` reports for that folder on your stick?

Anyhow, I think in the absence of further problems we can put this to bed.

**Reply by lyzanxia** [reply]
You can try with : `attrib +r f:\Photons-ADF-Demo-Disk-Archive\*.* /s` , which breaks it for me
And fix it again with `attrib -r f:\Photons-ADF-Demo-Disk-Archive\*.* /s` 

Where F is my usbstick obviously..

**Reply by keirf** [reply]
Oh wow! It's happened!! That's good, I guess. I will have to find some time to debug it now. :)

**Reply by lyzanxia** [reply]
Yay :-) .. thanks for thinking with me.

**Reply by keirf** [reply]
Ok it was a straightforward nasty bug introduced in v3.36. Any read-only file would provoke it when `interface=amiga`. Thanks for helping to track it down!

**Reply by keirf** [reply]
Could you try the firmware linked here: https://github.com/keirf/flashfloppy/suites/8835715192/artifacts/402639062

It has some other fixes too. You have to be logged into GitHub for the link to work.

Thanks!

**Reply by keirf** [reply]
Should now be fixed in v3.37. Please test.

**Reply by lyzanxia** [reply]
Yes, works perfect now. It's fixed. Thanks!
